project-docs/stumptown-writers-guide.md

```diff
@@ -0,0 +1,697 @@
+
+**Note: we will need to think about how to extend this grammar for more complex sidebars.**
+
+**Note: we should probably abandon this "infinite levels of sections" thing here and perhaps restrict ourselves to three levels (by not allowing children to contain children).**
+
+## Chapter lists
+
+A "chapter list" is just a way to provide a reusable list of links to pages, presented in a particular order. For example, a sidebar might want to list all the HTML guide pages in a curated order, with the most basic first, and more advanced concepts later. So the sidebar specification could provide its own list of pages to include. But an HTML landing page might also want to include the same list, in the same order. This is what chapter lists enable you to do.
+
+A chapter list is a YAML file with the following contents:
+
+-   `title`: a mandatory property whose value is a string. This provides a title for the list.
+-   `chapters`: an array of paths to pages to include, relative to the location of the chapter list itself.
+
+For example, the chapter list for HTML guides, which is at content/html/guides:
+
+```
+title: Guides
+chapters:
+    - Block-level_elements
+    - Inline_elements
+    - Link_types
+    - Applying_color
+    - Date_and_time_formats
+    - Preloading_content
+    - CORS_settings_attributes
+    - CORS_enabled_image
+    - DASH_Adaptive_Streaming_for_HTML_5_Video
+    - Microdata
+    - microformats
+    - Quirks_Mode_and_Standards_Mode
+```
+
+## Specifications
+
+Writers use the `specifications` front matter property to describe the specifications that define a particular item. But they just list URLs there, and we want the rendered output to include the name of the specification too.
+
+We could ask writers to include the specification name as well, but this would involve repetition and place a burden on the writers to get the name right. So instead we keep a separate file that maps specification URLs to specification names. This file is called "specifications.yaml" and is kept under content/data.
+
+Here's an example:
+
+```yaml
+# Allowed specification domains and their titles
+# Never add historical specfications to this allow list
+
+html.spec.whatwg.org: 'WHATWG HTML Living Standard'
+w3c.github.io/webappsec-cspee: 'Content Security Policy: Embedded Enforcement'
+w3c.github.io/webappsec-feature-policy: 'Feature Policy'
+wicg.github.io/priority-hints: 'Priority Hints'
+```
+
+Items listed in the `specifications` front matter property must match one of the URLs in this file. They may add extra path elements and fragments, to point to individual items within a specification:
+
+```
+specifications: https://html.spec.whatwg.org/multipage/semantics.html#the-abbr-element
+```
+
+**Note: I don't see a reason why the front matter property includes a scheme, but the scheme is not present in "specifications.yaml".**
+
+## Contributors
```

Comment:
Maybe we could not talk about this at all? Or at least say, "There's historic contributor data in `contributors.md`. Please don't modify these files."

Reply:
I've updated this section.